### DISCUSSION TITLE
FR: ability to query raw prices through AJAX

### DISCUSSION THREAD

**Discussion by el-schneider** [duncanmcclean/simple-commerce#747] [Feature Requests]
### What would you like to see added?

I think currently there is no option to obtain the raw prices (without currency signs) through AJAX, but it would be useful for calculations or having more control over how currency signs are displayed (e.g. before or after the amount)

Maybe some aditional fields would work like`grand_total_raw` or `shipping_total_raw`

or prices could be objects like
```js
price: {
   formatted: "\u20ac12.00",
   raw: 1200,
   currency: "EUR"
}
```

### Is there a workaround you can use in the meantime?

Sure, stripping of signs and parsing as numbers can be done to get something to calculate with. 

### Priority wise - where's this feature for you?

Nice to have

**Comment by duncanmcclean** [comment]
I don't believe there's any way for us to add extra fields like you've suggested without making a breaking change around how folks use prices in Antlers.

I'm guessing you're talking about when you get all line items from the cart - correct?

**Comment by el-schneider** [comment]
Maybe giving control over currency sign placement by currency might be an option then. There are a lot of different ways how countries handle the placement of their currency signs ([check here](https://fastspring.com/blog/how-to-format-30-currencies-from-countries-all-over-the-world/)) and being stuck with a preceding currency sign without space seems a bit limiting. Or is there an option I just haven't found yet?

**Reply by duncanmcclean** [reply]
Yeah, we could potentially have a way to disable the formatting from any prices that Simple Commerce returns so it would always give you the price in pence, then you'd have to figure out how to format it yourself.

Do you think that would work for your use case?

**Reply by el-schneider** [reply]
It's probably the cleanest way, as displaying of prices should be dealt with by the client anyways, because doing it properly is almost as complex as localizing dates

```js
const price = 1200;

let formatted = price.toLocaleString("de", { style: "currency", currency: "EUR" });
console.log(formatted);
// 1.200,00 €

formatted = price.toLocaleString("en", { style: "currency", currency: "USD" });
console.log(formatted);
// $1,200.00

formatted = price.toLocaleString("de", { style: "currency", currency: "USD" });
console.log(formatted);
// 1,200.00 $
```

**Reply by duncanmcclean** [reply]
Okay, I'll see what I can do over the weekend.

Simple Commerce uses PHP's currency formatting under the hood which isn't perfect but works 'okay-ish' for a decent number of currencies.

**Comment by duncanmcclean** [comment]
Hey!

I've just pushed out an update ([v4.2.0](https://github.com/duncanmcclean/simple-commerce/releases/tag/v4.2.0)) which takes advantage of Statamic's new 'Computed Values' feature meaning we can actually return a separate 'raw price'.

In fact, after updating, if you do an AJAX request, you should see a `raw_price` value.

Have a great rest of your weekend!
